_plugins/rosindex_generator.rb

```diff
@@ -300,18 +301,29 @@ def extract_package(site, distro, repo, snapshot, checkout_path, path, pkg_type,
       raw_uri = File.join(data['raw_uri'], relpath)
       browse_uri = File.join(data['browse_uri'], relpath)
 
-      # generate an array of pairs composed of the rendered and the not-rendered version
-      # of a readme file.
-      readmes_uncurated = readmes_relpath.map do |readme_relpath|
-        get_md_rst_txt(site, path, readme_relpath, raw_uri)
-      end.push(get_readme(site, path, raw_uri))
-
-      # curate the list by removing the nil elements
-      readmes_rendered, readmes = readmes_uncurated.reject do |x|
-        x[0].nil? || x[1].nil?
-      end.transpose
-      
-      # check for readme in same directory as package.xml
+      # load the package's readme for this branch if it exists
+      readme_file = Dir.glob(File.join(path, "README*"), File::FNM_CASEFOLD)
+      unless readme_file.empty? then
+        readmes_relpath.push(File.basename(readme_file.first))
+      end
+
+      # Iterate over each of the readme file paths that were explicitly declared in package
+      readmes = Array.new
+      readmes_relpath.each do |readme_relpath|
+        tmp_readme_rendered, tmp_readme  = get_md_rst_txt(site, path, readme_relpath, raw_uri)
+        readme = {
+          'browse_uri' => File.join(data['browse_uri'], package_name, readme_relpath),
+          'filename' => File.basename(readme_relpath),
```

Comment:
Could we actually pass the full filename here not just the Basename. Otherwise you have to introspect the url to find out which subdirectory it came from.

If I have
```
/
   README.md
   /package
      /README.md 
```
It would be good to know the difference in the rendering.

Reply:
For readme files found under a directory:
![screenshot from 2018-09-18 14-57-20](https://user-images.githubusercontent.com/5348967/45706687-45a63400-bb53-11e8-92d4-b1a65271701a.png)

For readme files found in the repository's root:
![screenshot from 2018-09-18 14-57-58](https://user-images.githubusercontent.com/5348967/45706686-45a63400-bb53-11e8-9c52-d85c1a7ec138.png)